packages/frontend/src/components/fields-editor/FieldsEditor.tsx

```diff
@@ -0,0 +1,215 @@
+          marginTop: "1rem",
+        }}
+      >
+        <CardContent>
+          <Grid2 container rowSpacing={2}>
+            <Grid2 size={12}>
+              <Typography gutterBottom variant="h6" component="div">
+                {title}
+              </Typography>
+            </Grid2>
+            <Grid2 size={12}>
+              <Divider />
+            </Grid2>
+            {values.map((e, idx) => (
+              <Filter
+                key={`filter-${idx}`}
+                value={e}
+                onChange={(newValue) => handdleChange(newValue, idx)}
+                onDelete={() => onDeleteElement(idx)}
+              />
+            ))}
+
+            <Grid2 size={12}>
+              <Button variant="text" onClick={handdleOnAddField} fullWidth>
+                <AddCircleOutlineIcon />
+              </Button>
+            </Grid2>
+          </Grid2>
+        </CardContent>
+      </Card>
+    </>
+  );
+};
+
+interface FilterProps {
+  value: HomeAssistantMatcher;
+  onChange: (value: HomeAssistantMatcher) => void;
+  onDelete: () => void;
+}
+
+const Filter = ({ value, onChange, onDelete }: FilterProps) => {
+  const handleOnChange = (newFieldValue: string, key: "type" | "value") => {
+    onChange({
+      ...value,
+      [key]: newFieldValue,
+    });
+  };
+  return (
+    <>
+      <Grid2 container spacing={{ md: 1, xs: 1.5 }} size={12}>
+        <Grid2 size={{ xs: 12, sm: 3, md: 2 }}>
+          <FormControl fullWidth>
+            <InputLabel id="select-field-type">Type</InputLabel>
+            <Select
+              onChange={(event) =>
+                handleOnChange(
+                  event.target.value as HomeAssistantMatcherType,
+                  "type",
+                )
+              }
+              labelId="select-field-type"
+              id="demo-simple-select"
+              value={value.type}
+              label="Age"
+            >
+              {Object.entries(HomeAssistantMatcherType).map((e) => (
+                <MenuItem value={e[1]} key={e[1]}>
+                  {e[0]}
+                </MenuItem>
+              ))}
+            </Select>
+          </FormControl>
+        </Grid2>
+        <Grid2 size={{ xs: 12, sm: 7, md: 9 }}>
+          <TextField
+            onChange={(event) =>
+              handleOnChange(event.target.value as string, "value")
+            }
+            label="Value"
+            variant="outlined"
+            value={value.value}
+            fullWidth
+          />
+        </Grid2>
+        <Grid2 size={{ xs: 12, sm: 2, md: 1 }}>
+          <Button
+            style={{ height: "100%", width: "100%", minHeight: 0 }}
+            size={"small"}
+            variant="outlined"
+            color="error"
+            onClick={onDelete}
+          >
+            <DeleteOutlineIcon />
+          </Button>
+        </Grid2>
+      </Grid2>
+      <Grid2 size={12}>
+        <Divider />
+      </Grid2>
+    </>
+  );
+};
+
+export default FieldsEditor;
```

Comment:
i'd split that whole file into multiple files. just to improve readability.
maybe also rename the components for better namings.